src/Document/CategoryManager.php

```diff
@@ -157,36 +106,85 @@ public function getPager(array $criteria, int $page, int $limit = 10, array $sor
         return $pager;
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getRootCategoryWithChildren(CategoryInterface $category): void
+    public function getRootCategoryWithChildren(CategoryInterface $category)
     {
-        throw new \RuntimeException('Not Implemented yet');
+        if (null === $category->getContext()) {
+            throw new \RuntimeException('Context cannot be null');
+        }
+        if (null !== $category->getParent()) {
+            throw new \RuntimeException('Method can be called only for root categories');
+        }
+
+        $context = $category->getContext();
+
+        $this->loadCategories($context);
+
+        foreach ($this->categories[$context->getId()] as $contextRootCategory) {
+            if ($category->getId() === $contextRootCategory->getId()) {
+                return $contextRootCategory;
+            }
+        }
+
+        throw new \RuntimeException('Category does not exist');
```

Comment:
```suggestion
        throw new \InvalidArgumentException('Category does not exist.');
```
BTW, is there a identifier we could use for the referenced category in these exception messages? Something like this:
>Category "ref" does not exist.